File: Caida_libre/Caida_libre.py
"""
Caida libre 

Luis Eduardo Sánchez González

Facultad de Ciencias Físico Matemáticas
Física Computacional
"""

import numpy as np
import sys

sys.path.append("../")

from PhyPy.EulerPy import *

class Sistema_mecanico(Euler):

	def __init__(self, gravedad):
	
		self.Posicion = []
		self.Velocidad = []
		self.tiempo = []
		self.g = gravedad

	def Ec_Diferencial(self, t, x, dx, y, dy, z, dz):

		return np.array([0,-9.8,0])

if __name__=="__main__":

	import matplotlib.animation as animation
	import matplotlib.pyplot as plt

	t, tf, dt, h = 0, 5, 0.01, 100 					 			
	
	posicion, velocidad = np.array([0,h,0]), np.array([0,0,0]) 

	Caida_Libre = Sistema_mecanico(9.81)
	Caida_Libre.Metodo_Euler_Cromer(t, tf, dt, posicion, velocidad)

	plt.title("Caida libre")
	plt.plot(Caida_Libre.tiempo, Caida_Libre.y, label = r"$y(t)$")
	plt.xlabel(r"tiempo (s)")
	plt.ylabel(r"Posicion (m)")
	plt.legend()
	plt.grid()
	plt.show()

File: Programas en C++/Caida_libre.cpp
/*
Caida libre (C++)

Luis Eduardo Sánchez González

Facultad de Ciencias Físico Matemáticas
Física Computacional
*/

#include<bits/stdc++.h>

using namespace std;

double Ecuacion_Dif(double y, double dy);

int main(){

	double t = 0, tf = 10, dt = 0.01;
	double l = 10, g = 9.81;

	int N = (tf-t)/dt, i = 0;

	double Posicion[N], Velocidad[N], tiempo[N];

	Posicion[0] = 1000;
	Velocidad[0] = 0;
		
	ofstream file_plot;
	file_plot.open("Solucion.txt");

	file_plot<<tiempo[i]<<"\t"<<Posicion[i]<<endl; 

	while(t < tf){

		double aceleracion = Ecuacion_Dif(Posicion[i], Velocidad[i]);

		Velocidad[i+1] = Velocidad[i] + (aceleracion)*dt;
		Posicion[i+1] = Posicion[i] + Velocidad[i+1]*dt;
		tiempo[i+1] = tiempo[i] + dt;
		
		file_plot<<tiempo[i]<<"\t"<<Posicion[i]<<endl;	

		t += dt;
		i++;

	}

	file_plot.close();

	system("python graficar.py");
			
	return 0;
}

double Ecuacion_Dif(double y, double dy){

	return -9.81;	
}
